tests/cases/conformance/es2019/globalThisAmbientModules.ts

```diff
@@ -0,0 +1,16 @@
+declare module "ambientModule" {
+    export type typ = 1
+    export var val: typ
+}
+namespace valueModule { export var val = 1 }
+namespace namespaceModule { export type typ = 1 }
+// should error
+type GlobalBad1 = (typeof globalThis)["\"ambientModule\""]
+type GlobalOk1 = (typeof globalThis)["valueModule"]
+type GlobalOk2 = globalThis.namespaceModule.typ
+const bad1: (typeof globalThis)["\"ambientModule\""] = 'ambientModule'
+
+type GlobalThis = keyof typeof globalThis
+// should error
+const bad2: GlobalThis = 'ambientModule'
```

Comment:
I'm not sure I understand what's being tested here, because the use of quotes makes it confusing for me. 
`const bad2: GlobalThis = 'ambientModule'` should error with or without this bug fix, no? Because the ambient module is `"ambientModule"`, with the quotes. Or am I misunderstanding something?

Reply:
This was the original test case, which was already confusing. I should probably just leave it out: it turns out that `keyof` turns `"\"foo\""` into just `"foo"`. But it's not relevant to whether `"\"foo\""` should be there in the first place.